pkg/cmd/pulumi-language-java/main.go

```diff
@@ -158,49 +160,55 @@ func (host *javaLanguageHost) Executor(attachDebugger bool) (*executors.JavaExec
 	return executor, nil
 }
 
-// GetRequiredPlugins computes the complete set of anticipated plugins required by a program.
-func (host *javaLanguageHost) GetRequiredPlugins(
+// GetRequiredPackages computes the complete set of anticipated packages required by a program.
+func (host *javaLanguageHost) GetRequiredPackages(
 	ctx context.Context,
-	req *pulumirpc.GetRequiredPluginsRequest,
-) (*pulumirpc.GetRequiredPluginsResponse, error) {
-	logging.V(5).Infof("GetRequiredPlugins: program=%v", req.GetProgram()) //nolint:staticcheck
-
-	// now, introspect the user project to see which pulumi resource packages it references.
-	pulumiPackages, err := host.determinePulumiPackages(ctx, req)
+	req *pulumirpc.GetRequiredPackagesRequest,
+) (*pulumirpc.GetRequiredPackagesResponse, error) {
+	pulumiPackages, err := host.determinePulumiPackages(ctx, req.Info.ProgramDirectory)
 	if err != nil {
 		return nil, errors.Wrapf(err, "language host could not determine Pulumi packages")
 	}
 
-	// Now that we know the set of pulumi packages referenced, and we know where packages have been restored to,
-	// we can examine each package to determine the corresponding resource-plugin for it.
-
-	plugins := []*pulumirpc.PluginDependency{}
+	pkgs := []*pulumirpc.PackageDependency{}
 	for _, pulumiPackage := range pulumiPackages {
-		logging.V(3).Infof(
-			"GetRequiredPlugins: Determining plugin dependency: %v, %v",
-			pulumiPackage.Name, pulumiPackage.Version,
-		)
-
+		// Skip over any packages that don't correspond to Pulumi resource plugins.
 		if !pulumiPackage.Resource {
-			continue // the package has no associated resource plugin
+			continue
 		}
 
-		plugins = append(plugins, &pulumirpc.PluginDependency{
+		pkg := &pulumirpc.PackageDependency{
+			Kind:    "resource",
 			Name:    pulumiPackage.Name,
 			Version: pulumiPackage.Version,
 			Server:  pulumiPackage.Server,
-			Kind:    "resource",
-		})
+		}
+		if pulumiPackage.Parameterization != nil {
+			pkg.Parameterization = &pulumirpc.PackageParameterization{
+				Name:    pulumiPackage.Parameterization.Name,
+				Version: pulumiPackage.Parameterization.Version,
+				Value:   pulumiPackage.Parameterization.Value,
+			}
+		}
+
+		pkgs = append(pkgs, pkg)
 	}
 
-	logging.V(5).Infof("GetRequiredPlugins: plugins=%v", plugins)
+	logging.V(5).Infof("GetRequiredPackages: packages=%v", pkgs)
+	return &pulumirpc.GetRequiredPackagesResponse{Packages: pkgs}, nil
+}
 
-	return &pulumirpc.GetRequiredPluginsResponse{Plugins: plugins}, nil
+// GetRequiredPlugins computes the complete set of anticipated plugins required by a program.
+func (host *javaLanguageHost) GetRequiredPlugins(
+	context.Context,
+	*pulumirpc.GetRequiredPluginsRequest,
+) (*pulumirpc.GetRequiredPluginsResponse, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method GetRequiredPlugins not implemented")
```

Comment:
I was wondering if we still need to implement this, so it works with an older engine.  But language runtime plugins are always bundled, so we know we'll always call GetRequiredPackages first anyway. Good.

Reply:
On both your comments, I've tried to match what was done for Go, NodeJS and Python -- it seems like we did the add/remove simultaneously for them and as you say, this is bundled so I _think_ it's OK?

Reply:
Yeah, I think this is fine, I just stumbled over it, so wanted to double check my thought process :)